kratos/utilities/embedded_skin_utility.cpp

```diff
@@ -205,6 +241,28 @@ namespace Kratos
         }
     }
 
+    template<std::size_t TDim>
+    Condition::Pointer EmbeddedSkinUtility<TDim>::SetNewConditionPointer(
+        const GeometryData::KratosGeometryType &rOriginGeometryType,
+        const Condition::NodesArrayType &rNewNodesArray,
+        const unsigned int &rConditionId,
+        const Properties::Pointer pConditionProperties)
+    {
+        const auto &r_condition = KratosComponents<Condition>::Get(this->GetConditionType());
+        auto p_new_geom = this->SetNewConditionGeometry(rOriginGeometryType, rNewNodesArray);
+        return r_condition.Create(rConditionId, p_new_geom, pConditionProperties);
+    }
+
+    template<>
+    const std::string EmbeddedSkinUtility<2>::GetConditionType() const {
```

Comment:
use kratos_components only once

Reply:
Done. Now a member reference to the prototype condition is used.